docs/book/beyond_basics/creating_events.md

```diff
@@ -1,26 +1,143 @@
+
+## Writing Events
+
+Anatomy of an Event:
+
 ```go
 Event(object runtime.Object, eventtype, reason, message string)
 ```
 
 - `object` is the object this event is about.
 - `eventtype` is the type of this event, and is either *Normal* or *Warning*.
-- `reason` is the reason this event is generated.  It should be short and unique with
-  `UpperCamelCase` format.   The value could appear in *switch* statements by automation.
+- `reason` is the reason this event is generated. It should be short and unique with `UpperCamelCase` format. The value could appear in *switch* statements by automation.
 - `message` is intended to be consumed by humans.
 
+
+Building on the example introduced in [Controller Example](../basics/simple_controller.md), we can add Events to our reconcile logic using `recorder` as our `EventRecorder`
+
+{% sample lang="go" %}
+```go
+var _ reconcile.Reconciler = &ContainerSetController{}
+
+func (r *ReconcileContainerSet) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+  instance := &workloadsv1beta1.ContainerSet{}
+  err := r.Get(context.TODO(), request.NamespacedName, instance)
+  if err != nil {
+    if errors.IsNotFound(err) {
+      // Object not found, return. Created objects are automatically garbage collected.
+      // For additional cleanup logic use finalizers.
+      return reconcile.Result{}, nil
+    }
+    // Error reading the object - requeue the request.
+    return reconcile.Result{}, err
+  }
+
+  // TODO(user): Change this to be the object type created by your controller
+  // Define the desired Deployment object
+  deploy := &appsv1.Deployment{
+    ObjectMeta: metav1.ObjectMeta{
+      Name:      instance.Name + "-deployment",
+      Namespace: instance.Namespace,
+    },
+    Spec: appsv1.DeploymentSpec{
+      Selector: &metav1.LabelSelector{
+        MatchLabels: map[string]string{"deployment": instance.Name + "-deployment"},
+      },
+      Replicas: &instance.Spec.Replicas,
+      Template: corev1.PodTemplateSpec{
+        ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"deployment": instance.Name + "-deployment"}},
+        Spec: corev1.PodSpec{
+          Containers: []corev1.Container{
+            {
+                Name:  instance.Name,
+                Image: instance.Spec.Image,
+            },
+          },
+        },
+      },
+    },
+  }
+  
+  if err := controllerutil.SetControllerReference(instance, deploy, r.scheme); err != nil {
+    return reconcile.Result{}, err
+  }
+
+  // TODO(user): Change this for the object type created by your controller
+  // Check if the Deployment already exists
+  found := &appsv1.Deployment{}
+  err = r.Get(context.TODO(), types.NamespacedName{Name: deploy.Name, Namespace: deploy.Namespace}, found)
+  if err != nil && errors.IsNotFound(err) {
+    log.Printf("Creating Deployment %s/%s\n", deploy.Namespace, deploy.Name)
+    err = r.Create(context.TODO(), deploy)
+    if err != nil {
+      return reconcile.Result{}, err
+    }
```

Comment:
We should probably get rid of the code above this line and replaces it with a few lines `....` so that usage of recorder is highlighted ?

Reply:
Changed